src/Symfony/Bundle/FrameworkBundle/Command/ServerCommand.php

```diff
@@ -41,4 +41,25 @@ protected function getLockFile($address)
     {
         return sys_get_temp_dir().'/'.strtr($address, '.:', '--').'.pid';
     }
+
+    protected function isOtherServerProcessRunning($address)
+    {
+        $lockFile = $this->getLockFile($address);
+
+        if (file_exists($lockFile)) {
+            return true;
+        }
+
+        list($hostname, $port) = explode(':', $address);
```

Comment:
I know that you reused this part from the original code of the command, but what about replacing this basic `list + explode` by a proper [parse_url() PHP function](http://php.net/manual/en/function.parse-url.php) to avoid any problem with URL parsing?